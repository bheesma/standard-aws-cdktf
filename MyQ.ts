import { Construct } from "constructs";
import { SqsQueue } from "@cdktf/provider-aws/lib/sqs-queue";
import  { AwsProvider } from "@cdktf/provider-aws/lib/provider";

export class MyQConstruct extends Construct {
    
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // define resources here

    new AwsProvider(this,'aws',{
      region: "ap-southeast-2"
    });

    const queueName = 'MyQueueConstruct';
    new SqsQueue(this, queueName, {
      name: queueName
    });



  }
}
