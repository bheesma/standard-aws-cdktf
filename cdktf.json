{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "679d25ee-31c0-4b01-9e78-f9e68dcaa0e4",
  "sendCrashReports": "false",
  "terraformProviders": [
    "hashicorp/aws@~> 4.60.0"
  ],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}