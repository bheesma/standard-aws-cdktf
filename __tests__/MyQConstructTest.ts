// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import "cdktf/lib/testing/adapters/jest"; // Load types for expect matchers
import { Testing } from "cdktf";
import { SqsQueue } from "@cdktf/provider-aws/lib/sqs-queue";
import { MyQConstruct } from "../MyQ";

describe("My CDKTF Construct", () => {


  describe("construct", () => {
    it("Construct", () => {
      expect(
        Testing.synthScope((scope) => {
          new MyQConstruct(scope, "app");
        })
      ).toHaveResourceWithProperties(SqsQueue, {
        name: "MyQueueConstruct"
      });
    })
  });

  describe("Unit testing using snapshots", () => {
    it("Tests the snapshot", () => {

      expect(Testing.synthScope((scope) => {
        new MyQConstruct(scope, "app");
      })).toMatchSnapshot();

    })
  });


});
