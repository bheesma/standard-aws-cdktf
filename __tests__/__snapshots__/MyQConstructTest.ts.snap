// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My CDKTF Construct Unit testing using snapshots Tests the snapshot 1`] = `
"{
  "provider": {
    "aws": [
      {
        "region": "ap-southeast-2"
      }
    ]
  },
  "resource": {
    "aws_sqs_queue": {
      "app_MyQueueConstruct_78111FA7": {
        "name": "MyQueueConstruct"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "4.60.0"
      }
    }
  }
}"
`;
